recipes/PeoplesSpeech/peoples_speech_prepare.py

```diff
@@ -0,0 +1,421 @@
+        Target subset. People's speech contains multiple subsets, which must be
+        loaded invidividually and then concatenated. E.g. 'clean', 'clean_sac',
+        'dirty' or 'dirty_sa'. E.g. to combine  ['clean', 'dirty'].
+    skip_prep : bool, optional
+        If True, the data preparation will be skipped, and the function will return immediately.
+
+    Returns
+    -------
+    None
+    """
+
+    if not os.path.isdir(hf_download_folder):
+        msg = "You must download the dataset with HuggingFace before starting "
+        msg += (
+            "this recipe. Please check the HuggingFace hub of people's speech."
+        )
+        raise ValueError(msg)
+
+    if skip_prep:
+        logger.info("Skipping data preparation as `skip_prep` is set to `True`")
+        return
+
+    if len(subsets) == 0:
+        raise ImportError(
+            "At least one People's speech subset must be specified."
+        )
+
+    # Setting output paths
+    save_output = {}
+    splits = ["train", "validation", "test"]
+    for split in splits:
+        save_output[split] = os.path.join(save_folder, str(split) + ".csv")
+
+    # check if the data is already prepared
+    if skip_csv(save_output):
+        logger.info("Skipping preparation, completed in previous run.")
+        return
+    else:
+        logger.info("Starting data preparation...")
+
+    hf_dataset = load_and_concatenate_datasets(subsets, hf_download_folder)
+
+    logger.info(
+        f"Preparing CSV of the Peoples Speech dataset in {save_folder}..."
+    )
+
+    os.makedirs(save_folder, exist_ok=True)
+
+    for i, split, output in enumerate(save_output.items()):
+        logger.info(f"Starting creating {output} using {split} split.")
+        HF_create_csv(output, hf_dataset[i], split)
+
+    logger.info("Data preparation completed!")
+
+
+def load_and_concatenate_datasets(subsets, hf_download_folder):
+    """Load/download and concatenate all the specified subsets from People's
+    speech. The people's speech dataset have 4 subset "clean", "clean_sa",
+    "dirty" and "dirty_sa". Multiple subsets cannot be loaded all at once with
+    HuggingFace so this function makes it possible.
+
+    Parameters
+    ----------
+    subsets : list
+        Target subset. People's speech contains multiple subsets, which must be
+        loaded invidividually and then concatenated. E.g. 'clean', 'clean_sac',
+        'dirty' or 'dirty_sa'. E.g. to combine  ['clean', 'dirty'].
+    hf_download_folder : str
+        The path where HF stored the dataset. Important, you must set the global
+        env variable HF_HUB_CACHE to the same path as HuggingFace is primilarily
+        using this to know where to store datasets.
+
+    Returns
+    -------
+    List of HuggingFace dataset.
+    """
+
+    try:
+        import datasets
+        from datasets import concatenate_datasets, load_dataset
+    except ImportError:
+        raise ImportError("HuggingFace datasets must be installed.")
```

Comment:
I think you shouldn't raise the exception like that. It can silently skip some errors (e.g. one missing package while you have datasets).

Reply:
In my case, I had a missing package in datasets while having datasets. So the ImportError was not helpful at all to debug this (i didn't had pyarrow for a Compute Canada reason :p)